### PR TITLE
Initialize continuous integration with Travis CI

### PR DESCRIPTION
**Pull request by kiudee**
# Description

To get the ball rolling with properly tested pull requests, this pull requests sets up Travis CI. It will set up a basic Miniconda installation, retrieve all necessary packages and run all tests.

# Discussion points:

- [x] Choice of LICENSE: How permissive do we want to be?
- [x] Choice of CI tool: Are there better alternatives?

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        description=DESCRIPTION,
+        author=AUTHOR,
+        author_email=AUTHOR_EMAIL,
+        packages=find_packages(),
+        install_requires=install_reqs,
+        include_package_data=True,
+        url=URL,
+        license=LICENSE
+    )
+)
```
Comment (Lambdanaut) [inline setup.py]: Additional parenthesis throwing syntax error.

Comment (kiudee) [reply setup.py]: Thanks @Lambdanaut, it should be fixed now.

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+numpy>=1.15
+PySC2>=2.0
```
Comment (nikola-j) [inline requirements.txt]: Maybe use the git version, the pip one is very outdated like so:
```
git+https://github.com/deepmind/pysc2.git@a9f093493c4c77adb385602790a480e7f238b97d
```

Comment (kiudee) [reply requirements.txt]: Good point, I will replace it

**Comment by inoryy** [comment]
What's the benefit of Apache license over MIT? And :+1: from me on Travis CI.

**Comment by Matuiss2** [comment]
I always advocate for very permissive licenses like MIT, but Apache is better for this project since it will receive a lot of contributors. This link explains it ver well
https://exygy.com/which-license-should-i-use-mit-vs-apache-vs-gpl/

**Comment by kiudee** [comment]
How can we persuade Travis to succeed, if there are no tests?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from os.path import realpath, dirname, join
+from setuptools import setup, find_packages
+
+DISTNAME = 'leelastar'
+DESCRIPTION = 'LeelaStar: Crowd-Sourced Deep Reinforcement Learning'
```
Comment (inoryy) [inline setup.py]: Are we going the general framework route? There's quite a few good solutions out there, so I think focusing on a niche of 1v1 StarCraft 2 would make more sense.

File: requirements.txt (hunk 2 above)
Comment (inoryy) [reply requirements.txt]: I guess we could add PyTorch as a dependency, unless somebody has strong preference again (voted 4:4 yesterday)

File: scripts/create_testenv.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+set -ex # fail on first error, print commands
+
+command -v conda >/dev/null 2>&1 || {
+  echo "Requires conda but it is not installed.  Run install_miniconda.sh." >&2;
+  exit 1;
+}
+
+ENVNAME="testenv"
+PYTHON_VERSION=${PYTHON_VERSION:-3.7} # if no python specified, use 3.7
```
Comment (inoryy) [inline scripts/create_testenv.sh]: If we're going to use TensorFlow then Python 3.7 could be an issue

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+AUTHOR = 'LeelaStar Developers'
+AUTHOR_EMAIL = 'leelastar.devs@gmail.com'
+URL = "https://github.com/leelastar/leelastar-training"
+LICENSE = "Apache License, Version 2.0"
+VERSION = "0.1"
```
Comment (inoryy) [inline setup.py]: 0.0.1? :)

**Comment by Lambdanaut** [comment]
I'm seeing an error with the create_env script on my macbook 

install_miniconda.sh worked fine, and it looks like running it this second time confirms it, but create_env complains that I don't have it installed.

```
Administrators-MacBook-Pro-2:leelastar-training jthomas$ sudo sh scripts/install_miniconda.sh 
Miniconda already installed at /Users/jthomas/miniconda3.  Updating, adding to path and exiting
Adding /Users/jthomas/miniconda3 to PATH.  Consider adding it in your .rc file as well.
Collecting package metadata: ...working... done
Solving environment: ...working... done

# All requested packages already installed.

Administrators-MacBook-Pro-2:leelastar-training jthomas$ sh scripts/create_testenv.sh 
+ command -v conda
+ echo 'Requires conda but it is not installed.  Run install_miniconda.sh.'
Requires conda but it is not installed.  Run install_miniconda.sh.
+ exit 1
```

File: scripts/create_testenv.sh (hunk 4 above)
Comment (Lambdanaut) [reply scripts/create_testenv.sh]: 5:4 in favor of pytorch actually (my vote was late)

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    - $HOME/miniconda3
+
+before_install:
+  - . ./scripts/install_miniconda.sh
+
+install:
+  - . ./scripts/create_testenv.sh
+
+script:
+  - pytest -v
```
Comment (inoryy) [inline .travis.yml]:
```suggestion
  - pytest
```

**Comment by islamelnabarawy** [comment]
@Lambdanaut Did you import conda into your bash session? You might need to source the right file for it to work.

**Comment by inoryy** [comment]
@Lambdanaut works for me on Ubuntu 18.04

**Comment by Lambdanaut** [comment]
Thanks guys. It installed to my home directory but didn't add that to my path, so I added it manually. Running now.

File: requirements.txt [hunk 7]
```diff
@@ -0,0 +1,2 @@
+numpy>=1.15
+-e git+https://github.com/deepmind/pysc2.git@a9f093493c4c77adb385602790a480e7f238b97d
```
Comment (islamelnabarawy) [inline requirements.txt]:
```suggestion
-e git+https://github.com/deepmind/pysc2.git@a9f093493c4c77adb385602790a480e7f238b97d#egg=PySC2
```

**Comment by Lambdanaut** [comment]
New to travis. How do I interact with travis.yml here? I installed travis using a package manager.

File: setup.py (hunk 3 above)
Comment (kiudee) [reply setup.py]: I am open to new suggestions.

What about “LeelaStar: Crowd-Sourced Deep Reinforcement Learning Agent for Starcraft 2”?

**Pull request merged**